eng/common/scripts/modules/Package-Properties/Package-Properties.psm1

```diff
@@ -0,0 +1,172 @@
+        pushd $RepoRoot
+        $setupProps = (python -c "import scripts.devops_tasks.common_tasks; obj=scripts.devops_tasks.common_tasks.parse_setup('$setupLocation'); print('{0},{1}'.format(obj[0], obj[1]));") -split ","
+        Write-Host $setupProps
+        popd
+        if (($setupProps -ne $null) -and ($setupProps[0] -eq $pkgName))
+        {
+            $pkgReadMePath = Join-Path $pkgPath "README.md"
+            $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+            $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+            $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+
+            return @{
+                pkgName = $setupProps[0];
+                pkgVersion = $setupProps[1];
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+function Extract-JavaPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "pom.xml"
+
+    if (Test-Path $projectPath)
+    {
+        $projectData = New-Object -TypeName XML
+        $projectData.load($projectPath)
+        $projectPkgName = Select-XML -Xml $projectData -XPath '/*[local-name()="project"]/*[local-name()="artifactId"]/text()'
+        $pkgVersion = Select-XML -Xml $projectData -XPath '/*[local-name()="project"]/*[local-name()="version"]/text()'
+
+        $pkgReadMePath = Join-Path $pkgPath "README.md"
+        $pkgReadMePath = If (Test-Path ($pkgReadMePath)) {  $pkgReadMePath } Else { $null }
+        $pkgChangeLogPath = Join-Path $pkgPath "CHANGELOG.md"
+        $pkgChangeLogPath = If (Test-Path ($pkgChangeLogPath)) {  $pkgChangeLogPath } Else { $null }
+        
+        if ($projectPkgName.ToString() -eq $pkgName)
+        {
+            return @{
+                pkgName = $pkgName;
+                pkgVersion = $pkgVersion.ToString();
+                pkgDirPath = $pkgPath;
+                pkgReadMePath = $pkgReadMePath;
+                pkgChangeLogPath = $pkgChangeLogPath;
+            }
+        }
+    }
+    return $null
+}
+
+# Takes package name and service Name
+# Returns important properties of the package as related to the language repo
+# Returns a HashTable with Keys @ { pkgName, pkgVersion, pkgDirPath, pkgReadMePath, pkgChangeLogPath }
+function Get-PkgProperties {
+    Param
+    (
+        [Parameter(Mandatory=$true)]
+        [string]$pkgName,
+        [Parameter(Mandatory=$true)]
+        [string]$serviceName
+    )
+
+    $pkgDirectoryPath = $null
+    $serviceDirectoryPath = Join-Path $RepoRoot "sdk" $serviceName
+    if (!(Test-Path $serviceDirectoryPath))
+    {
+        Write-Error "Service Directory $serviceName does not exist"
+        exit 1
+    }
+
+    $directoriesPresent = Get-ChildItem $serviceDirectoryPath -Directory
+
+    foreach ($directory in $directoriesPresent)
```

Comment:
Why are we looping over all the directories in the service folder when you are already passing the pkgName?

Reply:
Some package directories are named different from the package name in some cases. I initially added a switch statement that matches the directory name against regex patterns, that will prevent having to look to verify each package before determining the right one, but then the script will break if the directory naming convention is ignored in another case.

I can add that back but I wasn't sure if that was the best thing to do.

Reply:
I agree that the directory might differ but it looks like currently we will always get all the packages for a given directory. Did I miss some other filtering somewhere that is comparing the pkgname to what you parsed?

Reply:
So the Extract-{Language}PkgProps function only return if the supplied package Name matches the actual extracted package name property. Yes it does evaluate each package, but it returns null if its not the right package.